### PR TITLE
Add function read until 

### PR DESCRIPTION
**Pull request by sabas1080**
Hi @pelikhan 

I am trying to port the Serial read.Until to read the GPS data that is identified with the "$" delimiter, but it seems not to separate the lines with delimiter and also the reading stops after a few seconds

makecode maker blocks javascript editor 9

makecode maker blocks javascript editor 10

Thanks

### CODE REVIEW

File: libs/serial/serial.cpp [hunk 1]
```diff
@@ -48,6 +63,19 @@ SINGLETON_IF_PIN(WSerial,TX);
...
+    /**
+     * Read a line of text from the serial port and return the buffer when the delimiter is met.
+     * @param delimiter text delimiter that separates each text chunk
+     */
+    //% help=serial/read-until
+    //% blockId=serial_read_until block="serial|read until %delimiter=serial_delimiter_conv"
+    //% 
+    String readUntil(String delimiter) {
+      auto service = getWSerial();
+      auto s = service->serial.readUntil((delimiter->data, delimiter->length));
```
Comment (pelikhan) [inline libs/serial/serial.cpp]: double parenthesis?

File: libs/serial/serial.ts [hunk 2]
```diff
@@ -5,6 +5,18 @@
...
+    /**
+     * Read a line of text from the serial port and return the buffer when the delimiter is met.
+     * @param delimiter text delimiter that separates each text chunk
+     */
+    //% help=serial/read-line
+    //% blockId=serial_read_line block="serial|read line %delimiter=serial_delimiter_conv"
+    //% weight=19
+    //% group="Read"
+    export function readLine(delimiter: string): string {
+        return readUntil(delimiters(Delimiters.Dollar));
```
Comment (pelikhan) [inline libs/serial/serial.ts]: you should pass delimiter

**Comment by sabas1080** [comment]
Thanks @pelikhan , work delimiter, but "$" and "new line" two o two line and stop

makecode maker blocks javascript editor 12

I think the buffer is saturated and it stops working, 

with "," working

makecode maker blocks javascript editor 11

**Comment by sabas1080** [comment]
I have increased the buffer to 90 with "setRxBufferSize", as in this library

https://github.com/adafruit/GPS-shield-RMC-test-sketch/blob/master/GPStest_RMC.ino#L32

and work

makecode maker blocks javascript editor 13

File: libs/serial/serial.ts [hunk 3]
```diff
@@ -5,6 +5,18 @@
...
+    /**
+     * Read a line of text from the serial port and return the buffer when the delimiter is met.
+     * @param delimiter text delimiter that separates each text chunk
+     */
+    //% help=serial/read-line
+    //% blockId=serial_read_line block="serial|read line %delimiter=serial_delimiter_conv"
+    //% weight=19
+    //% group="Read"
+    export function readLine(delimiter: string): string {
+        return readUntil(delimiter);
```
Comment (pelikhan) [inline libs/serial/serial.ts]: this should be new line alway

File: libs/serial/serial.ts [hunk 4]
```diff
@@ -5,6 +5,18 @@
 //% advanced=true
 //% groups='["Write", "Read", "Events", "Configuration"]'
 namespace serial {
+    /**
+     * Read a line of text from the serial port and return the buffer when the delimiter is met.
+     * @param delimiter text delimiter that separates each text chunk
```
Comment (pelikhan) [inline libs/serial/serial.ts]: not needed

**Comment by pelikhan** [comment]
Superseeded by other PR. Thansk!

**Comment by sabas1080** [comment]
Thanks @pelikhan

**Pull request closed without merging**